importer/lambda/cross_region_importer.py

```diff
@@ -136,30 +136,45 @@ def _delete_cross_stack_references(exports_to_remove, table_info, physical_resou
                 raise
 
 
-@retry(
-    wait=wait_random_exponential(multiplier=2, max=30),
-    retry=retry_if_exception_type(ClientError),
-    stop=stop_after_delay(270)
-)
 def _get_cloudformation_exports(target_region, requested_exports):
     cloudformation_client = boto3.client('cloudformation', region_name=target_region)
-    paginator = cloudformation_client.get_paginator('list_exports')
-    exports_page_iterator = paginator.paginate()
 
     exports_tracker = {export_name: False for export_name in requested_exports.values()}
     exports = {}
-    
-    for page in exports_page_iterator:
-        for export in page['Exports']:
-            if export['Name'] in exports_tracker:
-                exports[export['Name']] = {'Value': export['Value'], 'ExportingStackId': export['ExportingStackId']}
-                exports_tracker[export['Name']] = True
-        if all(export_state for export_state in exports_tracker.values()): 
-            break
+
+    export_page = _retry_safe_list_exports(cloudformation_client)
+    _parse_exports(exports, exports_tracker, export_page)
+
+    if not all(export_state for export_state in exports_tracker.values()):
+        while 'NextToken' in export_page:
+            export_page = _retry_safe_list_exports(
+                cloudformation_client, next_token=export_page['NextToken'])
+            _parse_exports(exports, exports_tracker, export_page)
+
+            if all(export_state for export_state in exports_tracker.values()):
```

Comment:
No list comprehension needed here
```suggestion
            if all(exports_tracker.values()):
```

Reply:
Indeed!